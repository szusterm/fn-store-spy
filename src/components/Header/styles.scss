@import '~bootstrap/scss/functions';
@import '~bootstrap/scss/variables';
@import '~bootstrap/scss/mixins/breakpoints';

@import '~scss/_opacity-colors';
@import '~scss/_center-items';

.header {
	$padding-normal: 150px;
	$padding-mobile: 80px;

	@include center-items();
	flex-direction: column;

	overflow: hidden;
	position: relative;
	padding: $padding-normal 0 ($padding-normal/1.5);

	@include media-breakpoint-down(sm) {
		flex-direction: column;
		padding: $padding-mobile 0;
	}

	&::before {
		background-image: url('/images/background.jpg');
		background-size: cover;

		display: block;
		content: '';
		z-index: -1;

		position: absolute;
		min-width: 100%;
		top: 0;
		bottom: 0;

		transform: scale(1.1);

		overflow: hidden;
		filter: blur(10px) brightness(70%);
	}

	&--site-info {
		@include center-items();
		flex-direction: row;

		margin-bottom: 50px;

		@include media-breakpoint-down(sm) {
			flex-direction: column;
		}

		.site-info {
			&--logo {
				$size-normal: 140px;
				$size-mobile: 120px;

				width: $size-normal;
				height: $size-normal;

				@include media-breakpoint-down(sm) {
					width: $size-mobile;
					height: $size-mobile;
				}
			}

			&--text {
				font-size: 2em;
				font-weight: 100;
				margin-bottom: 0;
				margin-top: 20px;
				margin-left: 30px;

				@include white-color(0.9);

				@include media-breakpoint-down(sm) {
					text-align: center;
					margin-left: 0;
					font-size: 1.8em;
					padding: 5px;
				}

				&--bold {
					font-weight: 400;
				}
			}
		}
	}
}