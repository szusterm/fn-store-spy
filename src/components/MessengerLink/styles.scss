@import '~scss/_input';
@import '~scss/_transition';
@import '~scss/_center-items';

.messenger-link {
	$messenger-color: #0084ff;

	margin: 10px 0;

	&:hover {
		text-decoration: none;
	}

	&--button {
		@include default-input();
		@include center-items();
		flex-direction: row;

		border: 1px solid;
		border-color: white;
		background-color: $messenger-color;
		color: white;

		margin: 0;
		padding: 0 8px;

		@include transition(md);

		&:hover, &:focus {
			border-color: $messenger-color;
			background-color: white;
			color: $messenger-color;

			.button--svg-icon {
				.svg-icon--path {
					fill: $messenger-color;
				}
			}
		}

		.button {
			&--svg-icon {
				width: 34px;

				.svg-icon--path {
					fill: white;
				}
			}

			&--title {
				padding: 0 15px;

				font-size: 1.1em;
				font-weight: 400;
				text-align: center;
			}
		}

		&--reversed {
			border-color: $messenger-color;
			background-color: white;
			color: $messenger-color;

			&:hover, &:focus {
				border-color: white;
				background-color: $messenger-color;
				color: white;

				.button--svg-icon {
					.svg-icon--path {
						fill: white;
					}
				}
			}

			.button {
				&--svg-icon {
					.svg-icon--path {
						fill: $messenger-color;
					}
				}
			}
		}
	}
}