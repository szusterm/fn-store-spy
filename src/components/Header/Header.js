import React, {Component} from 'react';

import Logo from '../Logo';

import './styles.scss';

export class Header extends Component {
	render() {
		return (
			<header>
				<div className={'header'}>
					<div className={'header--logo-box'}>
						<Logo/>
					</div>
					<h1 className={'header--text'}>
						<span className={'header--text--bold'}>Hello. </span>
						Welcome to StoreSpy.<br/>
						Messenger shop notifier.
					</h1>
				</div>
			</header>
		);
	}
}

export default Header;