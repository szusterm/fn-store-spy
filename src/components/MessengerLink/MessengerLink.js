import React, {Component} from 'react';

import './styles.scss';

export class MessengerLink extends Component {
	render() {
		return (
			<div className={'col-3'}>
				<a
					className={'messenger-link'}
					href={'#'}
				>
					<div className={'messenger-link--button'}>
						m.me/ssbot
					</div>
				</a>
			</div>
		);
	}
}

export default MessengerLink;