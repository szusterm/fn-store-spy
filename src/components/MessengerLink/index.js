import MessengerLink from './MessengerLink';

export default MessengerLink;