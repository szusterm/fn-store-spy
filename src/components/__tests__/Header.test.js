import React from 'react';
import {shallow} from 'enzyme';

import MessengerLink from '../MessengerLink';

import {Header} from '../Header/Header';

const setup = (propOverrides) => {
	const props = Object.assign({}, propOverrides);

	const wrapper = shallow(<Header {...props}/>);

	return {props, wrapper};
};

describe('Header Component', () => {
	it('renders correctly', () => {
		const {wrapper} = setup();

		expect(wrapper).toMatchSnapshot();
	});

	it('renders MessengerLink', () => {
		const {wrapper} = setup();

		expect(wrapper.find(MessengerLink).exists()).toBe(true);
	});

	it('puts reversed prop to MessengerLink', () => {
		const {wrapper} = setup();

		const reversed = wrapper.find(MessengerLink).props().reversed;

		expect(reversed).toBe(true);
	});
});