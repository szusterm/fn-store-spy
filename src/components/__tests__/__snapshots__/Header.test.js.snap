// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header Component renders correctly 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Header />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <div
        className="header"
      >
        <div
          className="header--site-info"
        >
          <img
            className="site-info--logo"
            src="../images/logo.svg"
          />
          <h1
            className="site-info--text"
          >
            <span
              className="site-info--text--bold"
            >
              Hello. 
            </span>
            Welcome to StoreSpy.
            <br />
            Messenger shop notifier.
          </h1>
        </div>
        <Connect(MessengerLink)
          reversed={true}
        />
      </div>,
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <div
            className="header--site-info"
          >
            <img
              className="site-info--logo"
              src="../images/logo.svg"
            />
            <h1
              className="site-info--text"
            >
              <span
                className="site-info--text--bold"
              >
                Hello. 
              </span>
              Welcome to StoreSpy.
              <br />
              Messenger shop notifier.
            </h1>
          </div>,
          <Connect(MessengerLink)
            reversed={true}
          />,
        ],
        "className": "header",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <img
                className="site-info--logo"
                src="../images/logo.svg"
              />,
              <h1
                className="site-info--text"
              >
                <span
                  className="site-info--text--bold"
                >
                  Hello. 
                </span>
                Welcome to StoreSpy.
                <br />
                Messenger shop notifier.
              </h1>,
            ],
            "className": "header--site-info",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "className": "site-info--logo",
                "src": "../images/logo.svg",
              },
              "ref": null,
              "rendered": null,
              "type": "img",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  <span
                    className="site-info--text--bold"
                  >
                    Hello. 
                  </span>,
                  "Welcome to StoreSpy.",
                  <br />,
                  "Messenger shop notifier.",
                ],
                "className": "site-info--text",
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": "Hello. ",
                    "className": "site-info--text--bold",
                  },
                  "ref": null,
                  "rendered": "Hello. ",
                  "type": "span",
                },
                "Welcome to StoreSpy.",
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {},
                  "ref": null,
                  "rendered": null,
                  "type": "br",
                },
                "Messenger shop notifier.",
              ],
              "type": "h1",
            },
          ],
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "reversed": true,
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": "div",
    },
    "type": "header",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <div
          className="header"
        >
          <div
            className="header--site-info"
          >
            <img
              className="site-info--logo"
              src="../images/logo.svg"
            />
            <h1
              className="site-info--text"
            >
              <span
                className="site-info--text--bold"
              >
                Hello. 
              </span>
              Welcome to StoreSpy.
              <br />
              Messenger shop notifier.
            </h1>
          </div>
          <Connect(MessengerLink)
            reversed={true}
          />
        </div>,
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <div
              className="header--site-info"
            >
              <img
                className="site-info--logo"
                src="../images/logo.svg"
              />
              <h1
                className="site-info--text"
              >
                <span
                  className="site-info--text--bold"
                >
                  Hello. 
                </span>
                Welcome to StoreSpy.
                <br />
                Messenger shop notifier.
              </h1>
            </div>,
            <Connect(MessengerLink)
              reversed={true}
            />,
          ],
          "className": "header",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <img
                  className="site-info--logo"
                  src="../images/logo.svg"
                />,
                <h1
                  className="site-info--text"
                >
                  <span
                    className="site-info--text--bold"
                  >
                    Hello. 
                  </span>
                  Welcome to StoreSpy.
                  <br />
                  Messenger shop notifier.
                </h1>,
              ],
              "className": "header--site-info",
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "className": "site-info--logo",
                  "src": "../images/logo.svg",
                },
                "ref": null,
                "rendered": null,
                "type": "img",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": Array [
                    <span
                      className="site-info--text--bold"
                    >
                      Hello. 
                    </span>,
                    "Welcome to StoreSpy.",
                    <br />,
                    "Messenger shop notifier.",
                  ],
                  "className": "site-info--text",
                },
                "ref": null,
                "rendered": Array [
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {
                      "children": "Hello. ",
                      "className": "site-info--text--bold",
                    },
                    "ref": null,
                    "rendered": "Hello. ",
                    "type": "span",
                  },
                  "Welcome to StoreSpy.",
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {},
                    "ref": null,
                    "rendered": null,
                    "type": "br",
                  },
                  "Messenger shop notifier.",
                ],
                "type": "h1",
              },
            ],
            "type": "div",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "reversed": true,
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
        ],
        "type": "div",
      },
      "type": "header",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
